src/main/java/bio/terra/service/iam/sam/SamRetry.java

```diff
@@ -65,6 +66,8 @@ static void retry(ConfigurationService configService, SamVoidFunction function)
                 return function.apply();
             } catch (ApiException ex) {
                 handleApiException(ex);
+            } catch (IamUnauthorizedException ex) {
```

Comment:
This seems a bit odd to me. The exception is already being thrown in the next catch handler. The difference is the top level exception type; the wrapped exception is still present and could be used to generate the correct response. Or is there no way to change the response behavior based on a wrapped exception?

Reply:
As stated above, the `ApiException` being caught here is different from the `ApiException` being thrown by `AuthenticatedUserRequest::getRequiredToken`

Reply:
OK, that is confusing